packages/babel-helper-create-class-features-plugin/src/fields.ts

```diff
@@ -665,6 +666,24 @@ const thisContextVisitor = traverse.visitors.merge([
       state.needsClassRef = true;
       path.replaceWith(t.cloneNode(state.classRef));
     },
+    Function(path: NodePath<t.Function>, state) {
+      // only `() => { new.target }` needs to be replaced
+      state.shouldReplaceNewTarget = path.isArrowFunctionExpression();
+    },
+    MetaProperty(path: NodePath<t.MetaProperty>, state) {
+      const meta = path.get("meta");
+      const property = path.get("property");
+      const { scope } = path;
+      // if there are `new.target` in static field
+      // we should replace it with `undefined`
+      if (
+        state.shouldReplaceNewTarget &&
```

Comment:
I caused confusion using `state.shouldReplaceNewTarget` in my example. After seeing this complete code, `state.shouldReplaceNewTarget` is not needed: `environmentVisitor` (that is merged to this visitor) already skips non-arrow functions, we we don't need to have our custom `Function` visitor.

Reply:
Good point!